src/store/actions/actionCreators.js

```diff
@@ -0,0 +1,20 @@
+import { SIGNUP } from './types';
+
+export const signUp = user => async dispatch => {
+  await fetch('https://authors-haven-odin.herokuapp.com/api/users/', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json; charset=utf-8',
+    },
+    body: JSON.stringify({ user }),
+  })
+    .then(res => res.json())
```

Comment:
The first then of the `fetch` API only guarantees that the browser was able to make a request irrespective of what the result or status of the request is, it will resolve for both the success and error status codes . if you want to check if the request was successful, you have to check for the `ok` property and only proceed if it's true, if false, it would mean something went wrong , say there was a validation issue or you are hitting a  `non` existing `url` at which point you have to bail out, something like this.

```js
. then(res => res.json().then(data => (res.ok ? Promise.resolve(data) : Promise.reject(data))));
```
Clearly it's painful to do these sorts of checks every time you want to make a request, luckily , we have a wrapper that abstracts this sort of duplication in `src/utils/Http` and is injected in all our actions by `redux-thunk`, that means if you wanted, you could simplify all that code to something like this

```js
export const signUp = user => (dispatch, getState, http) =>
  http.post('users').then(data => dispatch({ type: SIGNUP, payload: data }));
```

Reply:
I currently have issues with using this wrapper, probably I will sync with you and you help me solve this issue